pages/agent/v3/cli_artifact.md

```diff
@@ -289,3 +290,75 @@ up, see our [Google Cloud installation guide](/docs/agent/v3/gcloud#uploading-ar
 You can configure the `buildkite-agent artifact` command to store artifacts in
 Artifactory. For instructions for how to set this up, see our
 [Artifactory guide](/docs/integrations/artifactory).
+
+## Using your private Azure Blob container
+
+>📘 Support for uploading artifacts to Azure Blob Storage was added in Agent
+> v3.53.0.
+
+You can configure the `buildkite-agent artifact` command to store artifacts in
+your private [Azure Blob Storage container](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction).
+
+### Preparation
+
+Firstly, make sure that each agent has access to Azure credentials.
+[By default](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#readme-defaultazurecredential)
+these can be provided using Azure environment variables such as
+`AZURE_CLIENT_ID`, loaded by a Kubernetes workload identity hook, loaded on a
+host with Azure Managed Identity enabled, or loaded from a user logged in with
+the Azure CLI.
+
+You can also use an account key or connection string by setting one of these
+environment variables:
+
+```shell
+# To use an account key:
+export BUILDKITE_AZURE_BLOB_ACCESS_KEY='...'
+
+# To use a connection string:
+export BUILDKITE_AZURE_BLOB_CONNECTION_STRING='...'
+```
+
+Since these can contain access credentials, they are
+[redacted from job logs by default](/docs/pipelines/managing-log-output#redacted-environment-variables).
+
+Make sure you have a valid storage account name and container. These can be
+created with the Azure web console or Azure CLI.
+
+Make sure the Azure principal for the Azure credential has a role assignment
+that permits reading and writing to the container, for example,
+[Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-contributor).
+
+### Configuration
+
+Configure the agent to target your container by exporting the
+`BUILDKITE_ARTIFACT_UPLOAD_DESTINATION` environment variable using an
+[environment agent hook](/docs/agent/v3/hooks) (this can not be set using the
+Buildkite web interface, API, or during pipeline upload). For example:
+
+```shell
+export BUILDKITE_ARTIFACT_UPLOAD_DESTINATION="https://mystorageaccountname.blob.core.windows.net/my-container/$BUILDKITE_PIPELINE_ID/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
+```
+
+Alternatively you can specify the full upload destination in the form
+`https://[storageaccountname].blob.core.windows.net/[container]/[path]` when
+running `buildkite-agent artifact upload` or `buildkite-agent artifact
+download`.
```

Comment:
I think it would be good to split out the URL form into a codeblock to make it easier for people to copy. We can rearrange the sentence to make that work:

````suggestion
Alternatively, when running `buildkite-agent artifact upload` or `buildkite-agent artifact
download`, you can specify the full upload destination in the form:

```
https://[storageaccountname].blob.core.windows.net/[container]/[path]
```
````